[3.12.0b4] Enum changes

The enum changes in `3.12` currently result in three test failures. (Tested with `3.12.0b4`)
```py
FAILED tests/test_quirks.py::test_read_attributes_uncached - TypeError: 'NoneType' object is not iterable
FAILED tests/test_zcl_clusters.py::test_time_cluster - TypeError: 'NoneType' object is not iterable
FAILED tests/test_zcl_clusters.py::test_ias_zone - TypeError: 'NoneType' object is not iterable
```

AFAICT it's related to how bitmap enums are deserialized. A simplified example
```py
import enum

class FixedIntType(int):
    def __new__(cls, *args, **kwargs):
        n = super().__new__(cls, *args, **kwargs)
        return n

class bitmap8(FixedIntType, enum.Flag):
    pass

print(bitmap8.from_bytes(b'\x00', 'little', signed=False))
```
https://github.com/zigpy/zigpy/blob/94a1603bd5b53b6bfa2781c13a9a0cf9f8e95c63/zigpy/types/basic.py#L195

#### Python 3.11.4
```py
bitmap8(0)
```

#### Python 3.12.0b4
```py
Traceback (most recent call last):
  File "/usr/src/test.py", line 14, in <module>
    print(bitmap8.from_bytes(b'\x00', 'little', signed=False))
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/enum.py", line 742, in __call__
    return cls._create_(
           ^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/enum.py", line 870, in _create_
    for item in names:
TypeError: 'NoneType' object is not iterable
```

I've opened https://github.com/python/cpython/issues/106928. The original author (of the cpython change) asked for more details why this specific pattern is necessary. It would be awesome if someone with knowledge about the implementation could reply there so we can get this resolved.
/CC @puddly

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -6,7 +6,8 @@ on:
 
 jobs:
   shared-ci:
-    uses: zigpy/workflows/.github/workflows/ci.yml@main
+    runs-on: ubuntu-latest
+    uses: zigpy/workflows/.github/workflows/ci.yml@puddly/pre-commit
     with:
       CODE_FOLDER: zigpy
       CACHE_VERSION: 2

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -10,5 +10,6 @@ jobs:
     with:
       CODE_FOLDER: zigpy
       CACHE_VERSION: 2
+      PRE_COMMIT_CACHE_PATH:  ~/.cache/pre-commit
       PYTHON_VERSION_DEFAULT: 3.9.15
       MINIMUM_COVERAGE_PERCENTAGE: 99

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -6,7 +6,6 @@ on:
 
 jobs:
   shared-ci:
-    runs-on: ubuntu-latest
     uses: zigpy/workflows/.github/workflows/ci.yml@puddly/pre-commit
     with:
       CODE_FOLDER: zigpy

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -6,7 +6,7 @@ on:
 
 jobs:
   shared-ci:
-    uses: zigpy/workflows/.github/workflows/ci.yml@puddly/pre-commit
+    uses: zigpy/workflows/.github/workflows/ci.yml@main
     with:
       CODE_FOLDER: zigpy
       CACHE_VERSION: 2

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -11,5 +11,4 @@ jobs:
       CODE_FOLDER: zigpy
       CACHE_VERSION: 2
       PYTHON_VERSION_DEFAULT: 3.9.15
-      PRE_COMMIT_CACHE_PATH:  ~/.cache/pre-commit
       MINIMUM_COVERAGE_PERCENTAGE: 99

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -46,7 +46,7 @@ testing = [
     "pytest-timeout",
     "freezegun",
     "ruff==0.0.261",
-    'pysqlite3-binary; platform_system=="Linux"',
+    'pysqlite3-binary; platform_system=="Linux" and python_version<"3.12"',
 ]
 
 [tool.setuptools-git-versioning]

--- a/requirements_test.txt
+++ b/requirements_test.txt
@@ -11,4 +11,4 @@ freezegun
 ruff==0.0.261
 tomli
 
-pysqlite3-binary; platform_system=="Linux"
+pysqlite3-binary; platform_system=="Linux" and python_version<"3.12"

--- a/zigpy/types/basic.py
+++ b/zigpy/types/basic.py
@@ -361,7 +361,14 @@ class uint64_t_be(uint_t_be, bits=64):
     pass
 
 
-class _IntEnumMeta(enum.EnumMeta):
+class AlwaysCreateEnumType(enum.EnumMeta):
+    """Enum metaclass that skips the functional creation API."""
+
+    def __call__(cls, value, names=None, *values) -> type[enum.Enum]:  # type: ignore
+        return cls.__new__(cls, value)  # type: ignore
+
+
+class _IntEnumMeta(AlwaysCreateEnumType):
     def __call__(cls, value, names=None, *args, **kwargs):
         if isinstance(value, str):
             if value.startswith("0x"):
@@ -382,7 +389,13 @@ def bitmap_factory(int_type: CALLABLE_T) -> CALLABLE_T:
 
     if sys.version_info >= (3, 11):
 
-        class _NewEnum(int_type, enum.ReprEnum, enum.Flag, boundary=enum.KEEP):
+        class _NewEnum(
+            int_type,
+            enum.ReprEnum,
+            enum.Flag,
+            boundary=enum.KEEP,
+            metaclass=AlwaysCreateEnumType,
+        ):
             pass
 
     else:

